### DISCUSSION TITLE
File Upload - long upload Filename Problem

### DISCUSSION THREAD

**Discussion by HS-cvift** [livewire/livewire#8287] [7. Bugs (archived)]
### Livewire version

v3.4.10

### Laravel version

v10.48.7

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. use File Upload with livewire
2. use simple html file upload dialog
3. upload a file with a very long file name (>200 chars, or fewer with special characters)
4. livewire cant store the temp file on disk, because the temp filename is too long

### Repository or Wirebox URL

n/a

### Additional information

problem is in file:
https://github.com/livewire/livewire/blob/main/src/Features/SupportFileUploads/TemporaryUploadedFile.php
function:
generateHashNameWithOriginalNameEmbedded($file)
line 162:
$meta = str('-meta'.base64_encode($file->getClientOriginalName()).'-')->replace('/', '_');

it produce a to long file name for the file system so it cant store the file

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by Naoray** [comment]
solved in https://github.com/livewire/livewire/pull/9172
